### DISCUSSION TITLE
What is joint?

### DISCUSSION THREAD

**Discussion by Qiandao-Liu** [empriselab/RCareWorld#55] [Help - Programming] [answered]
I was wondering what are joints for robot? is it the gripper position or the position between gripper and arm?

**Comment by YoruCathy** [comment]
Joint means the actuable parts on the robot. Every part that can move is a joint. For example the Stretch robot has a joint in the connector of the gripper and arm, and also multiple joints on the arm itself that can expand and contract.

**Reply by Qiandao-Liu** [reply]
got it, but when I want to expand the arm, I can't find joint_id, I try to print but the joint_id is not 0 to 8 (if there are 8 movable joints), is there any ways to get the joint_id for different part of the robot? (like the joint_id for gripper, and joints for arm expand)

**Reply by YoruCathy** [reply, marked as answer]
The joint position is basically a list with num_joint elements, each stands for a joint position for a joint, starting from the robot base. It's not like a robot that has its own ID.

**Reply by Qiandao-Liu** [reply]
hi, sorry for interupt, but I spent tons of work try to rotate a joint but it just doesn't work. Could you show me a demo command or what to let a gripper joint rotate for 90 degrees?
